Several fixes: Assume account + account list refresh + account creation

JBWatenbergScality opened the pull request:
Three issues fixed within this pull request: 

- Refresh list account post creation

- Creation of account enter issue

- Assume role issue when refresh page rapidly

JBWatenbergScality committed:
Partial fix

bert-e commented:
# Hello jbwatenbergscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko-ui/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-357 contains:


* `1.6.16`

* `2.0.7`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.6.16`

* `2.0.8`


Please check the `Fix Version/s` of ZKUI-357, or the target
branch of this pull request.

JBWatenbergScality committed:
try to fix assume role issue

ChengYanJin committed:
ZKUI-357: Refetch accounts query after account creation and deletion

ChengYanJin committed:
ZKUI-357: Fix the account creation can't be summited by enter

ChengYanJin committed:
ZKUI-357: Fix the Access denied when list objects

Root cause:
The current assumed account is not the one indicate in the URL.

1. Add a new condition in the useMemo of DataServiceRoleProvider to make sure the account specify in the URL is one going to assume role.
2. Since the DataServiceRoleProvider is outside of Route, we can't get the accountName from useParams, hence useRouteMatch.

Issue: ARTESCA-8562

ChengYanJin committed:
ZKUI-357: Fix the toogle of object versionning

MonPote committed:
ZKUI-358 - prevent queries to trigger if iamclient not ready

JBWatenbergScality commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-357 contains:


* `1.6.16`

* `2.0.7`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.6.16`

* `2.0.8`


Please check the `Fix Version/s` of ZKUI-357, or the target
branch of this pull request.




*The following options are set:* **approve**

MonPote commented:
ping

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-357 contains:


* `1.6.16`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.6.16`

* `2.0.8`


Please check the `Fix Version/s` of ZKUI-357, or the target
branch of this pull request.




*The following options are set:* **approve**

MonPote commented:
ping

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZKUI-357` into
`development/1.6`
* [w/2.0/bugfix/ZKUI-357](https://github.com/scality/zenko-ui/compare/development/2.0...w/2.0/bugfix/ZKUI-357) will be merged into `development/2.0` 



The following branches will **NOT** be impacted:

* `development/1.4`
* `development/1.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.6`

* :heavy_check_mark: `development/2.0`



The following branches will **NOT** be impacted:

* `development/1.4`
* `development/1.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.6`

* :heavy_check_mark: `development/2.0`



The following branches have **NOT** changed:

* `development/1.4`
* `development/1.5`



Please check the status of the associated issue ZKUI-357.

Goodbye jbwatenbergscality.
